src/App.js

```diff
@@ -3,33 +3,52 @@ import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
 function App() {
-	const [todoList, setTodoList] = useState()
+	const [todoList, setTodoList] = useState([])
 	const [isLoading, setIsLoading] = useState(true)
 
-	useEffect(() => {
-		new Promise((resolve, reject) => {
-			setTimeout(() => {
-				resolve({
-					data: {
-						todoList: JSON.parse(localStorage.getItem('savedTodoList') || []),
-					},
-				})
-			}, 2000)
-		})
-			.then((result) => {
-				setTodoList(result.data.todoList)
-				setIsLoading(false)
-			})
-			.catch((error) => {
-				console.log(error)
+	const fetchData = async () => {
+		const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+		console.log(url)
+
+		const options = {
+			method: 'GET',
+			headers: {
+				Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+			},
+		}
+		try {
+			const response = await fetch(url, options)
+			if (!response.ok) {
+				const message = `Error has ocurred: ${response.status}`
+				throw new Error(message)
+			}
+			const data = await response.json()
+
+			const todos = data.records.map((todo) => {
+				const newTodo = {
+					id: todo.id,
+					title: todo.title,
+				}
+
+				return newTodo
 			})
+			setTodoList(todos)
```

Comment:
This works just fine for now. However, it can lead to some unexpected bugs. Because React expects a new reference within the setter function when you want to trigger a re-render, it is possible that it could see this as the same reference and not trigger the re-render. This isn't a bug I've ever actually run into but to be safe I recommend passing your new array as: `setTodoList([...todos]).

Now you guarantee a new reference to your setter function, triggering a re-render.

Reply:
I did work on this and if you could check it again that would be greatly appreciated!